File: 238/main.go
package main

func productExceptSelf(nums []int) []int {
	l := len(nums)
	left_operand, right_operand := make([]int, l), make([]int, l)
	left_operand[0], right_operand[l-1] = 1, 1

	for i := 1; i < l; i++ {
		left_operand[i] = left_operand[i-1] * nums[i-1]
		right_operand[l-i-1] = right_operand[l-i] * nums[l-i]
	}
	for i := range nums {
		nums[i] = left_operand[i] * right_operand[i]
	}
	return nums
}

File: 238/main.py
class Solution:
    def productExceptSelf(self, nums: list[int]) -> list[int]:
        left, right = [1]*len(nums), [1]*len(nums)
        for i in range(1, len(nums)):
            left[i] = left[i-1]*nums[i-1]
            right[len(nums)-i-1] = right[len(nums)-i]*nums[len(nums)-i]
        for i in range(0, len(nums)):
            nums[i] = left[i]*right[i]

        return nums
